src/lens/EphemeralAllPositions.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Enumerable.sol";
+import "./PositionLens.sol";
+
+/// @notice A lens for Uniswap v3 that peeks into the current state of all positions by an owner without deployment
+/// @author Aperture Finance
+/// @dev The return data can be accessed externally by `eth_call` without a `to` address or internally by catching the
+/// revert data, and decoded by `abi.decode(data, (PositionState[]))`
+contract EphemeralAllPositions is PositionLens {
+    constructor(INPM npm, address owner) {
+        PositionState[] memory positions = allPositions(npm, owner);
+        bytes memory returnData = abi.encode(positions);
+        assembly ("memory-safe") {
+            // The return data in a constructor will be written to code, which may exceed the contract size limit.
+            revert(add(returnData, 0x20), mload(returnData))
```

Comment:
Does this mean in our NodeJS code, we need to catch the error when we `eth_call` the deployment?

Reply:
No. A reverted call is still a successful RPC. Ethers throws an error when abi decoding fails.

Reply:
SG. Thanks for the explanation!